mysql-test/suite/innodb/t/temp_truncate.test

```diff
@@ -32,3 +32,20 @@ connection default;
 COMMIT;
 SELECT COUNT(*) FROM t1;
 DROP TABLE t1;
+
+--echo # MDEV-33101 Server crashes when starting the server with
+--echo # innodb-force-recovery=6 and enabling the
+--echo # innodb_truncate_temporary_tablespace_now variable
+
+--let $restart_parameters=--innodb-force-recovery=6
+--source include/restart_mysqld.inc
+SHOW VARIABLES LIKE "innodb_read_only";
+SET GLOBAL innodb_truncate_temporary_tablespace_now=1;
+
+--let $restart_parameters=--innodb-read-only
+--source include/restart_mysqld.inc
+SHOW VARIABLES LIKE "innodb_read_only";
+SET GLOBAL innodb_truncate_temporary_tablespace_now=1;
+
+--let $restart_parameters=
+--source include/restart_mysqld.inc
```

Comment:
The test already does one normal restart at the very start, I guess to start up with an empty temporary tablespace. If we cover the `innodb_force_recovery=6` case before that restart, then we can avoid adding this one more ‘cleanup restart’ at the end.

A server restart can easily consume 1 or 2 seconds of additional time on each test run, and possibly more on slower builders. That is why I want to avoid them.

Reply:
I prefer to keep the functional tests together but I see the point. Made the changes in https://github.com/MariaDB/server/pull/2981 11.3 branch.

Reply:
Yes, this is a deviation from what I would consider good style, but I think that the tests are primarily for computers, not humans. Humans can find the context with `git blame` and `git show`.